### PR TITLE
afflib: fix 3.7.10 to have better dependencies

### PR DESCRIPTION
**Pull request by umireon**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

* Use macOS's expat
* Enable S3 support since expat and curl are always present
* Include its python binding as optional
* Ensure to link with OSXFUSE if HOMEBREW_PREFIX is not /usr/local
* Ensure not to link with OSXFUSE

### CODE REVIEW

File: Formula/afflib.rb [hunk 1]
```diff
@@ -15,25 +15,33 @@ class Afflib < Formula
...
-  depends_on :osxfuse => :optional
+
   depends_on "openssl"
 
+  depends_on "python" => :optional
+  depends_on :osxfuse => :optional
+
   # This patch fixes a bug reported upstream over there
   # https://github.com/simsong/AFFLIBv3/issues/4
   patch :DATA
```
Comment (MikeMcQuaid) [inline Formula/afflib.rb]: Would you consider sending this patch as a PR (as requested on that now-closed issue)? Thanks!

**Review by MikeMcQuaid — APPROVED** [review]

File: Formula/afflib.rb [hunk 2]
```diff
@@ -57,21 +65,24 @@ def install
...
 +++ b/configure.ac
-@@ -241,10 +241,6 @@ AC_ARG_ENABLE(fuse,
+@@ -238,13 +238,6 @@ AC_ARG_ENABLE(fuse,
  if test "x${enable_fuse}" = "xyes" ; then
    AC_MSG_NOTICE([FUSE requested])
    CPPFLAGS="-D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=26 $CPPFLAGS"
 -  if test `uname -s` = Darwin ; then
 -    AC_MSG_NOTICE([FUSE IS NOT SUPPORTED ON MACOS])
 -    enable_fuse=no
 -  fi
```
Comment (umireon) [inline Formula/afflib.rb]: This part is already merged on master: sshock/AFFLIBv3#13

Comment (MikeMcQuaid) [inline Formula/afflib.rb]: Great. Can we use https://github.com/sshock/AFFLIBv3/pull/13.patch as a patch URL instead?

**Comment by umireon** [comment]
Thank you for reviewing, @MikeMcQuaid 
sshock/AFFLIBv3#13 is included as a patch.

**Comment by MikeMcQuaid** [comment]
@umireon Sorry, I mean it would be good to submit everything in `patch :DATA` on this formula as PRs. I know you didn't create them but it would be great if you could submit them.

**Comment by umireon** [comment]
I am working on this but I don't think these two part won't be valid on the upstream because they remove the fallback on the error.
* https://github.com/umireon/homebrew-core/blob/843b528e3187c5b8f2ec55c8cd9bb667d78ac82f/Formula/afflib.rb#L67
* https://github.com/umireon/homebrew-core/blob/843b528e3187c5b8f2ec55c8cd9bb667d78ac82f/Formula/afflib.rb#L79-L81

**Comment by MikeMcQuaid** [comment]
@umireon Cool, thanks. I think we'll need to try and find something that's valid/mergeable upstream or just remove the relevant patch.

**Comment by umireon** [comment]
@MikeMcQuaid IMO the configure script of afflib is too intrusive and fuzzy, and therefore we can make a PR to fix that.
(e.g. It never fail for lack of libraries, plus it force to use the libraries under /opt/local if exists)
But making changes on configure.ac should take some time of me, because I am not familiar with autotools 😿

**Comment by umireon** [comment]
The patch is reconstructed.
Yet another PR for the attached patch should be created but after sshock/AFFLIBv3#14, because this conflicts with the current master.

**Comment by umireon** [comment]
@MikeMcQuaid Now I have sent all patches as PR and they are all merged: sshock/AFFLIBv3#13, sshock/AFFLIBv3#14, sshock/AFFLIBv3#15
Currently this formula have three patches but they will be able to be removed with the next release of afflibv3.

**Comment by MikeMcQuaid** [comment]
@umireon Fantastic work, I'm really impressed!

**Comment by umireon** [comment]
Thank you for merging, @MikeMcQuaid !

**Pull request closed without merging**